x/mongo/driver/topology/errors.go

```diff
@@ -69,11 +70,17 @@ func (e ServerSelectionError) Unwrap() error {
 
 // WaitQueueTimeoutError represents a timeout when requesting a connection from the pool
 type WaitQueueTimeoutError struct {
-	Wrapped                      error
-	PinnedCursorConnections      uint64
-	PinnedTransactionConnections uint64
-	maxPoolSize                  uint64
-	totalConnectionCount         int
+	Wrapped                  error
+	pinnedConnections        *pinnedConnections
+	maxPoolSize              uint64
+	totalConnectionCount     int
+	availableConnectionCount int
```

Comment:
Optional: Appending "Count" is generally redundant when the type is an integer. Consider removing "Count" so the names are just `totalConnections` and `availableConnections`.